be/src/agent/publish_version.cpp

```diff
@@ -27,6 +27,7 @@ struct TabletPublishVersionTask {
     int64_t partition_id{0};
     int64_t tablet_id{0};
     int64_t version{0};
+    int64_t current_version{0};
```

Comment:
What the differenct between version and current_version?

Reply:
version is an input parameter(request publish version), it should not be reused as an output param(tablet's current version). In the original code, this introduced a bug that dropped tablets' current version will be wrongly assigned  to requested publish version.

Reply:
found that max_continuous_version is already for this purpose, but not used in earlier code, so fix it.